pkg/netlight/resource/resource.go

```diff
@@ -0,0 +1,381 @@
+		}
+
+		if err := setLinkAddr(infPublic, ndmzIP); err != nil {
+			return fmt.Errorf("couldn't set link addr for public interface in namespace %s: %w", nsName, err)
+		}
+		if err := options.SetIPv6Forwarding(true); err != nil {
+			return fmt.Errorf("failed to enable ipv6 forwarding in namespace %q: %w", nsName, err)
+		}
+
+		if privateNet != nil {
+			privGwIp := privateNet.IP.To4()
+			// this is to take the first IP of the private range
+			// and use it as the gw address for the entire private range
+			privGwIp[net.IPv4len-1] = 1
+			// this IP is then set on the private interface
+			privateNet.IP = privGwIp
+			if err := setLinkAddr(infPrivate, privateNet); err != nil {
+				return fmt.Errorf("couldn't set link addr for private interface in namespace %s: %w", nsName, err)
+			}
+		}
+
+		// if err := setLinkAddr(infPrivate, )
+		if err := netlink.RouteAdd(&netlink.Route{
+			Gw: ndmzGwIP.IP,
+		}); err != nil && !os.IsExist(err) {
+			return fmt.Errorf("failed to add ndmz routing")
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	rules, err := nftRules.Open("nft/rules.nft")
+	if err != nil {
+		return nil, fmt.Errorf("failed to load nft.rules file")
+	}
+
+	if err := nft.Apply(rules, nsName); err != nil {
+		return nil, fmt.Errorf("failed to apply nft rules for namespace '%s': %w", name, err)
+	}
+
+	return &Resource{name}, setupMycelium(netNS, infMycelium, seed)
+}
+
+func Delete(name string) error {
+	nsName := fmt.Sprintf("n%s", name)
+	netNS, err := namespace.GetByName(nsName)
+	if errors.Is(err, os.ErrNotExist) {
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+
+	if err := destroyMycelium(netNS, zinit.Default()); err != nil {
+		return err
+	}
+
+	if err := namespace.Delete(netNS); err != nil {
+		return err
+	}
+
+	privateNetBr := fmt.Sprintf("r%s", name)
+	myBr := fmt.Sprintf("m%s", name)
+
+	if err := bridge.Delete(privateNetBr); err != nil {
+		return err
+	}
+
+	return bridge.Delete(myBr)
+
+}
+
+func setLinkAddr(name string, ip *net.IPNet) error {
+	link, err := netlink.LinkByName(name)
+	if err != nil {
+		return fmt.Errorf("failed to set link address: %w", err)
+	}
+
+	// if err := options.Set(name, options.IPv6Disable(false)); err != nil {
+	// 	return fmt.Errorf("failed to enable ip6 on interface %s: %w", name, err)
+	// }
+
+	addr := netlink.Addr{
+		IPNet: ip,
+	}
+	err = netlink.AddrAdd(link, &addr)
+	if err != nil && !os.IsExist(err) {
+		return fmt.Errorf("failed to add ip address to link: %w", err)
+	}
+
+	return netlink.LinkSetUp(link)
+}
+
+// Get return resource handler
+func Get(name string) (*Resource, error) {
+	nsName := fmt.Sprintf("n%s", name)
+
+	if namespace.Exists(nsName) {
+		return &Resource{name}, nil
+	}
+
+	return nil, fmt.Errorf("resource not found: %s", name)
+}
+
+// myceliumIpSeed is 6 bytes
+func (r *Resource) AttachPrivate(id string, vmIp net.IP) (device pkg.TapDevice, err error) {
```

Comment:
I can't figure out if this an idempotent function or not. But it's usually important that such functions to be idempotent so recreation doesn't happen if the service was restarted for example